examples/mosquitto/README.md

```diff
@@ -0,0 +1,225 @@
+# User authentication in Mosquitto using SVIDs with DNS names
+
+This is an example of how **spiffe-helper** can be used to authenticate users to
+a **Mosquitto** broker using x509 SVIDs.
+
+Mosquitto can be configured to use the CN of a certificate as a username to
+authenticate the client. This is done by setting the following two options in
+the `mosquitto.conf` file:
+```
+require_certificate true
+use_identity_as_username true
+```
+
+## Guide
+
+### Preqrequisites
+
+We will do this test using three virtual machines (VMs) running Ubuntu 22.04 on
+the same virtual network. The setup will be as follows:
+| hostname | IP | Description |
+| --- | --- | --- |
+| node0 | 10.211.55.2 | SPIRE server |
+| node1 | 10.211.55.20 | Mosquitto broker |
+| node2 | 10.211.55.21 | Mosquitto client |
+
+### 1. Configure and Run Spire Server on Node0
+
+Download or build the latest version of SPIRE and extract it to `/opt/spire` on
+`node0`. Edit the `conf/server/server.conf` file to set the `bind_address` to
+`10.211.55.2` and the `trust_domain` to `example.org`. The rest of the
+configuration can be left as is.
+
+After editing the configuration file should look like this:
+```
+server {
+    bind_address = "10.211.55.2"
+    bind_port = "8081"
+    socket_path = "/tmp/spire-server/private/api.sock"
+    trust_domain = "example.org"
+    data_dir = "./.data"
+    log_level = "DEBUG"
+}
+
+plugins {
+    DataStore "sql" {
+        plugin_data {
+            database_type = "sqlite3"
+            connection_string = "./.data/datastore.sqlite3"
+        }
+    }
+
+    NodeAttestor "join_token" {
+        plugin_data {
+        }
+    }
+
+    KeyManager "memory" {
+        plugin_data = {}
+    }
+
+    UpstreamAuthority "disk" {
+        plugin_data {
+            key_file_path = "./conf/server/dummy_upstream_ca.key"
+            cert_file_path = "./conf/server/dummy_upstream_ca.crt"
+        }
+    }
```

Comment:
Is the upstream authority necessary?

Reply:
Not sure if it is necessary, but this is the default that comes with the package. I just left is as is. I figured it is better to leave as much as possible the default.

Reply:
Got it. Still think its better to remove because then we need instructions on how to create those certs and they are not relevant to this example.